ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/CommandParser.java

```diff
@@ -0,0 +1,228 @@
+        final int closingToken = sql.indexOf(SINGLE_QUOTE, i + 1);
+        current += sql.substring(i, closingToken + 1);
+        i = closingToken;
+      } else if (i < sql.length() - 1 && sql.startsWith(SHORT_COMMENT_OPENER, i)) {
+        i = sql.indexOf(SHORT_COMMENT_CLOSER, i + 1);
+      } else if (i < sql.length() - 1 && sql.startsWith(LONG_COMMENT_OPENER, i)) {
+        i = sql.indexOf(LONG_COMMENT_CLOSER, i + 1) + 1;
+      } else if (sql.charAt(i) == SEMICOLON) {
+        current += ";";
+        commands.add(current);
+        current = "";
+      } else {
+        current += sql.charAt(i);
+      }
+    }
+
+    return commands;
+  }
+
+  /*
+  * Converts an expression into a Java object.
+  **/
+  public static Object toFieldType(final Expression expressionValue) {
+    if (expressionValue instanceof StringLiteral) {
+      return ((StringLiteral) expressionValue).getValue();
+    } else if (expressionValue instanceof IntegerLiteral) {
+      return ((IntegerLiteral) expressionValue).getValue();
+    } else if (expressionValue instanceof LongLiteral) {
+      return ((LongLiteral) expressionValue).getValue();
+    } else if (expressionValue instanceof DoubleLiteral) {
+      return ((DoubleLiteral) expressionValue).getValue();
+    } else if (expressionValue instanceof BooleanLiteral) {
+      return ((BooleanLiteral) expressionValue).getValue();
+    } else if (expressionValue instanceof DecimalLiteral) {
+      return ((DecimalLiteral) expressionValue).getValue();
+    } else if (expressionValue instanceof CreateArrayExpression) {
+      return ((CreateArrayExpression) expressionValue)
+          .getValues()
+          .stream()
+          .map(val -> toFieldType(val)).collect(Collectors.toList());
+    } else if (expressionValue instanceof CreateMapExpression) {
+      final Map<Object, Object> resolvedMap = new HashMap<>();
+      ((CreateMapExpression) expressionValue).getMap()
+          .forEach((k, v) -> resolvedMap.put(toFieldType(k), toFieldType(v)));
+      return resolvedMap;
+    } else if (expressionValue instanceof CreateStructExpression) {
+      final Map<Object, Object> resolvedStruct = new HashMap<>();
+      ((CreateStructExpression) expressionValue)
+          .getFields().stream().forEach(
+              field -> resolvedStruct.put(field.getName(), toFieldType(field.getValue())));
+      return resolvedStruct;
+    }
+    throw new IllegalStateException("Expression type not recognized: "
+        + expressionValue.toString());
+  }
+
+  /*
+  * Determines the type of command a sql string is and returns a SqlCommand.
+  **/
+  private static SqlCommand transformToSqlCommand(final String sql) {
+    final List<String> tokens = Arrays
+        .stream(sql.toUpperCase().split(QUOTED_STRING_OR_WHITESPACE))
+        .filter(s -> !s.isEmpty())
+        .collect(Collectors.toList());
+    if (tokens.get(0).equals(INSERT)
+        && tokens.get(1).equals(INTO)
+        && !tokens.get(3).equals(SELECT)
```

Comment:
What is this line for? If we're trying to rule out `INSERT INTO ... AS SELECT`, we should be checking for the token with index 4 (not 3), right?

Reply:
`INSERT INTO .. SELECT` doesn't have `AS`